### PR TITLE
support fastify 3.x - node >=10

### PR DESCRIPTION
**Pull request by mashpie**
Thanks a lot for this module! 

We've been using it successfully with fastify 2.x. Now as I was starting to check for upgrades and migration options I found fastify-hemera sticking to `^2.0.0`. Turned out to be working without change in 3.2.x. This PR is basically just changing that version to `>=2.0.0`.

But as I wanted proper tests I added some notes to readme and upgraded fastify and related dependencies. This introduced a new version of `server` which dropped support for node 8:

```js
error semver@7.3.2: The engine "node" is incompatible with this module. Expected version ">=10". Got "8.17.0"
206error Found incompatible module.
```

That's why this PR also changes node support and covers versions from 10 to 14 in travis.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,19 +42,19 @@
...
-    "hemera-plugin": "^2.0.0",
+    "fastify": "^3.2.0",
+    "hemera-plugin": "^2.0.2",
     "hemera-testsuite": "^4.1.0",
-    "light-my-request": "^3.2.0",
+    "light-my-request": "^4.0.1",
     "np": "^4.0.2",
     "pre-commit": "^1.2.2",
     "prettier": "^1.16.4",
     "tap-spec": "^5.0.0"
```
Comment (mashpie) [inline package.json]: tap-spec could be removed... but generally I don't mess with given options other than incremental :)

File: README.md [hunk 2]
```diff
@@ -64,3 +64,23 @@ fastify.route({
   }
 })
 ```
+
+## Test
+
+Testsuite makes use of [hemera-testsuite](https://github.com/hemerajs/hemera-testsuite), so your setup should meet it's prerequisits, esp.:
+
+A nats server must be installed on your machine [nats.io](https://nats.io/download/nats-io/nats-server/) and executable as `gnatsd`. So add `nats-server` as `gnatsd` to your `PATH` as alias, link whatever works. CHeck if it's available:
```
Comment (StarpTech) [inline README.md]: Small typo `CHecks`

File: index.js [hunk 3]
```diff
@@ -41,6 +41,6 @@ function fastifyHemera(fastify, opts, next) {
 }
 
 module.exports = fp(fastifyHemera, {
-  fastify: '^2.0.0',
+  fastify: '>=2.0.0',
```
Comment (StarpTech) [inline index.js]: I'd set it to >=3

File: package.json (hunk 1 above)
Comment (StarpTech) [reply package.json]: Feel free to remove it.

File: package.json [hunk 4]
```diff
@@ -42,19 +42,19 @@
...
   },
   "dependencies": {
-    "fastify-plugin": "^1.5.0",
-    "nats": "1.2.2",
-    "nats-hemera": "^7.0.0",
-    "tap": "^12.5.3"
+    "fastify-plugin": "^2.2.0",
+    "nats": "1.4.12",
+    "nats-hemera": "^7.0.2",
+    "tap": "^14.10.8"
```
Comment (StarpTech) [inline package.json:56]: This is a dev dep.

**Review by StarpTech — APPROVED** [review]
Great PR thanks!

File: package.json (hunk 4 above)
Comment (mashpie) [reply package.json:56]: Sure, reason for upgrade is that tests are using those too... at least fastify-plugin needs an upgrade to support fastify 3 - new tap comes with built in spec reporter

File: index.js (hunk 3 above)
Comment (mashpie) [reply index.js]: That'd break 2.x installs (which is still LTS) - So if you plan to... I'd suggest a major version bumb. Leaving that up to you. But fastify-hemera as it is, supports both.

File: package.json (hunk 1 above)
Comment (mashpie) [reply package.json]: removed

File: README.md (hunk 2 above)
Comment (mashpie) [reply README.md]: fixed

File: index.js (hunk 3 above)
Comment (StarpTech) [reply index.js]: For v2 you can still use the previous version. I will do a major bump.

Comment (mashpie) [reply index.js]: agreed... here you go

**Comment by StarpTech** [comment]
@mashpie Would you mind creating a simple Github Action to enable tests on push? It's also ok to do it afterward.

**Comment by mashpie** [comment]
Well, yes I could look into this. To me this looks a bit out-of-scope of the current PR. So for now I think the existing travis setup is just fine and working https://travis-ci.com/github/uscreen/fastify-hemera/builds/179037444

**Comment by StarpTech** [comment]
Alright 👍

**Comment by StarpTech** [comment]
@mashpie The new version was released https://github.com/hemerajs/fastify-hemera/releases/tag/v10.0.0

**Pull request merged**